macros/src/bundle.rs

```diff
@@ -79,23 +79,30 @@ fn gen_bundle_impl(
     let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
     let with_static_ids_inner = quote! {
         {
-            let mut tys = [#((::std::mem::align_of::<#tys>(), ::std::any::TypeId::of::<#tys>())),*];
+            let mut tys = [#((::core::mem::align_of::<#tys>(), ::core::any::TypeId::of::<#tys>())),*];
             tys.sort_unstable_by(|x, y| {
-                ::std::cmp::Ord::cmp(&x.0, &y.0)
+                ::core::cmp::Ord::cmp(&x.0, &y.0)
                     .reverse()
-                    .then(::std::cmp::Ord::cmp(&x.1, &y.1))
+                    .then(::core::cmp::Ord::cmp(&x.1, &y.1))
             });
-            let mut ids = [::std::any::TypeId::of::<()>(); #num_tys];
-            for (id, info) in ::std::iter::Iterator::zip(ids.iter_mut(), tys.iter()) {
+            let mut ids = [::core::any::TypeId::of::<()>(); #num_tys];
+            for (id, info) in ::core::iter::Iterator::zip(ids.iter_mut(), tys.iter()) {
                 *id = info.1;
             }
             ids
         }
     };
     let with_static_ids_body = if generics.params.is_empty() {
-        quote! {
-            static ELEMENTS: ::std::sync::OnceLock<[::std::any::TypeId; #num_tys]> = ::std::sync::OnceLock::new();
-            f(ELEMENTS.get_or_init(|| #with_static_ids_inner))
+        if cfg!(feature = "std") {
+            quote! {
+                static ELEMENTS: ::core::sync::OnceLock<[::core::any::TypeId; #num_tys]> = ::core::sync::OnceLock::new();
```

Comment:
Does this path (starting with `::core`) really exist for `OnceLock`?

More generally, does it make sense to keep both code paths? The `spin`-based one should work fine on `std`-enabled targets so it might improve maintainability to just have that one code path. Furthermore, you might want to use `spin::lazy::Lazy` instead of `spin::once::Once` as it is more targeted towards this particular use case.

Reply:
Sorry this is a typo... it should be `::std`.

I've removed it and used `spin::Laz for both std and no-std cases. The `hecs-macros/std` macro is also removed.